feature-libs/order/core/facade/reorder-order.service.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { CartModificationList } from '@spartacus/cart/base/root';
+import { Command, CommandService, CommandStrategy } from '@spartacus/core';
+import { ReorderOrderFacade } from '@spartacus/order/root';
+import { Observable } from 'rxjs';
+import { ReorderOrderConnector } from '../connectors/reorder-order.connector';
+
+@Injectable()
+export class ReorderOrderService implements ReorderOrderFacade {
+  protected reorderCommand: Command<
+    { orderId: string; userId: string },
+    CartModificationList
+  > = this.commandService.create<
+    { orderId: string; userId: string },
+    CartModificationList
+  >(
+    ({ orderId, userId }) =>
+      this.reorderOrderConnector.reorder(orderId, userId),
+    {
+      strategy: CommandStrategy.CancelPrevious,
+    }
+  );
+
+  constructor(
+    protected commandService: CommandService,
+    protected reorderOrderConnector: ReorderOrderConnector
+  ) {}
+
+  /**
+   * Schedule a replenishment order
+   */
+  reorder(orderId: string, userId: string): Observable<CartModificationList> {
```

Comment:
See comment in the component, the reorder command should likely not take a userid param, but rather fetch it from here.